quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/SyncFunction.java

```diff
@@ -113,16 +115,22 @@ public void setColumnsToMigrate(Set<String> columnsToMigrate) {
 		this.insertExpressions = ImmutableMap.copyOf(expressions);
 		this.updateExpressions = ImmutableMap.copyOf(insertExpressions);
 
-		this.updateIdentitiesForInserts = ImmutableMap.copyOf((Map<? extends String, ? extends String>)targetTable.getPrimaryKeyColumns().stream()
+		this.updateIdentitiesForInserts = ImmutableMap.copyOf((Map<? extends String, ? extends String>) targetTable.getColumns().stream()
+				.filter(column -> reverseLookup(mapping, column.getName()) != null)
```

Comment:
Newly made columns with the add-column migration get filtered here, without it an error happens